### ISSUE TITLE
Showing totals for stacked bar charts

### ISSUE THREAD

**Issue by tlrdstd** [plotly/plotly.js#85]
I've been trying to show totals on my stacked bar charts. I'd prefer to show them as an integral part of the graph, but I'd settle for hoverinfo if I had to.

This jsfiddle using annotations is the closest I've come: http://jsfiddle.net/u6f0q14n/3/

![](https://cloud.githubusercontent.com/assets/194300/11659874/70e8f35e-9d98-11e5-98d3-ed37c998a9cf.PNG)

That works, until I click the legend to hide a given trace:

![](https://cloud.githubusercontent.com/assets/194300/11659914/9acc5ec2-9d98-11e5-85f2-f9b610169281.PNG)

The annotations (totals) are left hanging in the air, and the graph does not nicely resize to the updated data. As best I can tell, the provided config options do not offer a pre-packaged way to solve this.

**Comment by PPPW** [comment]
I looked at #34, I wonder will it show total in 'markers+text' for stacked bar? It seems for me, if I specify 'text' for each series in stacked bar, then it will display text near each series, not sure is it possible to display a total number for all series, using 'markers+text'? Thanks!

**Comment by nicolaskruchten** [comment]
I'd also like the ability to show totals in hovertemplates.

Perhaps we want a `texttemplate` ?

**Comment by nicolaskruchten** [comment]
this seems like a stack-level rather than trace-level thing. related to #3614 ?

**Comment by mangel2095** [comment]
I had the same need, and maybe it's far to being the best solution, but it works, I used `restyleData ` property like input in the callback, and additionaly put the figure of the plot like state, in this way, with de `restyleData`, detect when user interact with the figure. I save in a list the legends that are `visible` in the data figure, then, when define annotations only do it with the visibles legends.

 Basically is recalculate all,  but at least that happens in a short time lapse.

**Comment by sean-wallace** [comment]
This would be a great feature for stacked area charts too!

**Comment by gvwilson** [comment]
Hi - this issue has been sitting for a while, so as part of our effort to tidy up our public repositories I'm going to close it. If it's still a concern, we'd be grateful if you could open a new issue (with a short reproducible example if appropriate) so that we can add it to our stack. Cheers - @gvwilson
